src/utils/index.js

```diff
@@ -0,0 +1,67 @@
+const { filterType, filterComponents } = require('./filters');
+const suffixes = require('./suffixes');
+const directionals = require('./directionals');
+
+function checkRoutes (routes, validation, options = {}) {
+  if (!routes) {
+    return null;
+  }
+
+  const { directional, parseRoute } = options;
+  const splitRoute = routes.trim().split(' ');
+
+  /*
```

Comment:
A lot going on from lines 18-41; let's try to dissect it;

```
// Given E Ridge Pl
const splitRoute = routes.trim().split(' ') // ['E', 'Ridge', 'Pl']

// Objective: Validate split routes against directions
const directions = [ ] // list of directionals
const suffix = [] // list of USPS suffixes

// determine if given route is suffix or directional
function fundType(route) {}

// standardize into something that is digestable
const newArray = splitRoutes.map((route) => { 
const type = findType(route)
return {
type,
route
} // returns { directional/suffix: route }
});

// parse and return digestables 

```

This is all pseudo code but this will bypass all the complicated logic below. `splitRoute.length - 1` and `splitRoute.length - 2` seems a bit random as well- Where do these numbers come from?

Reply:
I'll give it a look, and I'll let you know if anything. 

In regards to the `splitRoute.length - 1` and `splitRoute.length - 2`, some addresses have the suffix at the end, but on the occasions that it has a postdirectional, it will be the next to last... might be a bit convoluted, but at the time, it was what made sense to me.